File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;

/**
 * 品牌模块服务层接口
 */
public interface BrandService {

    /**
     * 查询所有品牌数据
     * @return
     */
    public List<TbBrand> findAll();

    /**
     * 查询返回分页列表
     * @param pageNum   --当前页码
     * @param pageSize  --每页数量
     * @return
     */
    public PageResult<TbBrand> findByPage(int pageNum,int pageSize);

    /**
     * 增加品牌数据
     * @param brand
     */
    public void addBrand(TbBrand brand);

    /**
     * 根据id查询品牌
     * @param id
     * @return
     */
    public TbBrand findOne(Long id);

    /**
     * 修改品牌信息
     * @param brand
     */
    public void updateById(TbBrand brand);

    /**
     * 删除
     */
    public void deleteByIds(Long[] ids);

    /**
     * 条件查询返回分页列表
     * @param pageNum   --当前页码
     * @param pageSize  --每页数量
     * @return
     */
    public PageResult findByPage(TbBrand brand,int pageNum,int pageSize);

}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//品牌服务层
app.service('brandService',function($http){
    //读取列表数据绑定到表单中
    this.findAll=function(){
        return $http.get('../brand/findAll.do');
    }

    //分页查询方法
    this.findByPage = function (pageNum, pageSize) {
       return $http.get("../brand/findByPage.do?pageNum=" + pageNum + "&pageSize=" + pageSize);
    }

    //根据id查询品牌信息
    this.findOne=function (id) {
        return $http.get("../brand/findOne.do?id="+id);
    }

    //增加
    this.addBrand=function(entity){
        return  $http.post('../brand/addBrand.do',entity );
    }
    //修改
    this.update=function(entity){
        return  $http.post('../brand/update.do',entity );
    }

    //批量删除操作
    this.dele=function (selectIds) {
        return $http.get("../brand/delete.do?ids="+selectIds);
    }

    //条件查询
    this.search=function (pageNum, pageSize,searchEntity) {
        return $http.post("../brand/search.do?pageNum=" + pageNum + "&pageSize=" + pageSize ,searchEntity);
    }

});
